api/src/BcGov.Malt.Web/Services/ProjectExtensions.cs

```diff
@@ -27,40 +25,71 @@ public static void AddProjectAccess(this IServiceCollection services, IConfigura
-                            HttpClient httpClient = httpClientFactory.CreateClient(projectResourceKey + "-authorization");
-
-                            // create the handler that will authenticate the call and add authorization header
-                            ITokenCache tokenCache = serviceProvider.GetRequiredService<ITokenCache>();
-                            ITokenService tokenService = new OAuthTokenService(new OAuthClient(httpClient), tokenCache);
-                            var handler = new TokenAuthorizationHandler(tokenService, CreateOAuthOptions(projectResource));
-                            return handler;
-                        });
+                        case ProjectType.SharePoint:
+                            ConfigureSharePoint(services, project, projectResource);
+                            break;
+                    }
                 }
             }
         }
 
+        private static void ConfigureDynamics(IServiceCollection services, ProjectConfiguration project, ProjectResource projectResource)
+        {
+            Debug.Assert(services != null, "Required ServiceCollection is null");
+            Debug.Assert(project != null, "Required ProjectConfiguration is null");
+            Debug.Assert(projectResource != null, "Required ProjectResource is null");
+            Debug.Assert(projectResource.Type == ProjectType.Dynamics, "Project type must be Dynamics");
+
+            string projectResourceKey = project.Id + "-dynamics";
+
+            // add authorization HttpClient 
+            services.AddHttpClient(projectResourceKey + "-authorization", configure => configure.BaseAddress = projectResource.AuthorizationUri)
+                ;
+
+            // add odata HttpClient 
+            // note: I do not like this IoC anti-pattern where we are using the service locator directly, however,
+            //       there are many named dependencies. There may be an opportunity to address this in the future
+            services.AddHttpClient(projectResourceKey, configure => configure.BaseAddress = projectResource.Resource)
+                .AddHttpMessageHandler((serviceProvider) =>
+                {
+                    // build the token service that talk to the OAuth endpoint 
+                    IHttpClientFactory httpClientFactory = serviceProvider.GetRequiredService<IHttpClientFactory>();
+                    HttpClient httpClient = httpClientFactory.CreateClient(projectResourceKey + "-authorization");
+
+                    // create the handler that will authenticate the call and add authorization header
+                    ITokenCache tokenCache = serviceProvider.GetRequiredService<ITokenCache>();
+                    ITokenService tokenService = new OAuthTokenService(new OAuthClient(httpClient), tokenCache);
+                    var handler = new TokenAuthorizationHandler(tokenService, CreateOAuthOptions(projectResource));
+                    return handler;

```

Comment:
nit/suggestion to combine these two lines into one: 
`return new TokenAuthorizationHandler(tokenService, CreateOAuthOptions(projectResource));`

Reply:
The reason why I split these in many cases, is I want to set break point after the object is created so I can view / examine the object.  It is possible to do that in the debugger without this, but it is not straight forward.